packages/app/src/containers/ThemeTextInspector.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  CursorTextIcon,
+  LinkBreak2Icon,
+  PlusIcon,
+  ResetIcon,
+  UpdateIcon,
+} from '@radix-ui/react-icons';
+import { Button, Select, Spacer } from 'noya-designsystem';
+import { Selectors } from 'noya-state';
+import { memo, useCallback, useMemo } from 'react';
+import { useTheme } from 'styled-components';
+import * as InspectorPrimitives from '../components/inspector/InspectorPrimitives';
+import {
+  useApplicationState,
+  useSelector,
+} from '../contexts/ApplicationStateContext';
+import useShallowArray from '../hooks/useShallowArray';
+
+const NO_TEXT_STYLE = 'none';
+
+export default memo(function ThemeTextInspector() {
+  const [, dispatch] = useApplicationState();
+
+  const iconColor = useTheme().colors.icon;
+
+  const sharedTextStyles = useShallowArray(
+    useSelector(Selectors.getSharedTextStyles),
+  );
+  const selectedLayers = useShallowArray(
+    useSelector(Selectors.getSelectedLayers),
+  );
+
+  const selectedTextStyleId = useMemo(
+    () =>
+      !selectedLayers.every(
```

Comment:
We should have a helper for this pattern since we use it so often: if every value in an array is the same, we return that value, otherwise undefined

I originally wanted `getMultiValue` to handle this case, e.g: 

```ts
getMultiValue(selectedLayers.map(v => v.id))
``` 

I remember we changed `getMultiValue` to return a `boolean`... but I'm not sure if the boolean-returning version is really an improvement over just calling `every` directly. Maybe we change `getMultiValue` to return the `T | undefined`, and change the other usages back to `.every`?

Reply:
I thinking we could have two functions because I do use the getMultiValue with boolean a lot, maybe a different functions.

Reply:
Ok, we can try that. Let's do it in a separate issue like you suggested